pkgs/by-name/sc/screenly-cli/package.nix

```diff
@@ -0,0 +1,54 @@
+{ darwin
+, fetchpatch
+, fetchFromGitHub
+, lib
+, perl
+, pkg-config
+, openssl
+, rustPlatform
+, stdenv
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "screenly-cli";
+  version = "0.2.3";
+
+  src = fetchFromGitHub {
+    owner = "screenly";
+    repo = "cli";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-rQK1EYb1xYtcxq0Oj4eY9PCFMoaYinr42W8NkG36ps0=";
+  };
+
+  cargoPatches = [
+    (fetchpatch {
+      url = "https://github.com/Screenly/cli/commit/898bd2e5e3a9653e3c3dde17e951469885734c40.patch";
+      hash = "sha256-Cqc1PHRhgS3zK19bSqpU2v+R3jSlOY6oaLJXpUy6+50=";
+      includes = [ "Cargo.lock" ];
+    })
+  ];
```

Comment:
Could you explain in a comment why we need this patch and when we can remove it? _This is a general piece of advice_
In that particular situation, I have read through the patch and I am not even sure to understand why we need it, given that is only make changes to `gitignore`, `flake.nix`, and `cargo.lock` which should not have impact on this derivation? If the change is necessary because it introduces the cargo lock you can also vendor it directly in nixpkgs by using `cargoLock.lockFile = ...`.

Reply:
I realize you've already been ask to change this to use `fetchpatch`, which mean I'll be happy if you just put a comment explaining why you're applying this particular patch and a link to the PR.

Reply:
Just pushed a fix that introduces a comment. I rebased to keep the history neat on merge.